rabbitmq/content.md

```diff
@@ -29,50 +29,46 @@ This will start a RabbitMQ container listening on the default port of 5672. If y
 
 Note the `database dir` there, especially that it has my "Node Name" appended to the end for the file storage. This image makes all of `/var/lib/rabbitmq` a volume by default.
 
-### Memory Limits
-
-RabbitMQ contains functionality which explicitly tracks and manages memory usage, and thus needs to be made aware of cgroup-imposed limits.
-
-The upstream configuration setting for this is `vm_memory_high_watermark`, and it is described under ["Memory Alarms"](https://www.rabbitmq.com/memory.html) in the documentation.
-
-In this image, this value is set via `RABBITMQ_VM_MEMORY_HIGH_WATERMARK`. The value of this environment variable is interpreted as follows:
-
--	`0.49` is treated as `49%`, just like upstream (`{ vm_memory_high_watermark, 0.49 }`)
--	`56%` is treated as `56%` (`0.56`; `{ vm_memory_high_watermark, 0.56 }`)
--	`1073741824` is treated as an absolute number of bytes (`{ vm_memory_high_watermark, { absolute, 1073741824 } }`)
--	`1024MiB` is treated as an absolute number of bytes with a unit (`{ vm_memory_high_watermark, { absolute, "1024MiB" } }`)
-
-The main behavioral difference is in how percentages are handled. If the current container has a memory limit (`--memory`/`-m`), a percentage value will be calculated to an absolute byte value based on the memory limit, rather than being passed to RabbitMQ as-is. For example, a container run with `--memory 2048m` (and the implied upstream-default `RABBITMQ_VM_MEMORY_HIGH_WATERMARK` of `40%`) will set the effective limit to `819MB` (which is `40%` of `2048MB`).
-
-### Erlang Cookie
+### Environment Variables
 
-See the [RabbitMQ "Clustering Guide"](https://www.rabbitmq.com/clustering.html#erlang-cookie) for more information about cookies and why they're necessary.
+For a list of environment variables supported by RabbitMQ itself, see the [Environment Variables section of rabbitmq.com/configure](https://www.rabbitmq.com/configure.html#supported-environment-variables)
 
-For setting a consistent cookie (especially useful for clustering but also for remote/cross-container administration via `rabbitmqctl`), use `RABBITMQ_ERLANG_COOKIE`:
+**WARNING:** As of RabbitMQ 3.9, all of the docker-specifc variables listed below are deprecated and no longer used. Please use a configuration file instead; visit [rabbitmq.com/configure](https://www.rabbitmq.com/configure.html) to learn more about the configuration file. For a starting point, the 3.8 images will print out the config file it generated from supplied environment variables.
```

Comment:
```diff
- docker-specifc
+ docker-specific
```

Reply:
I think that we should encourage users to use configuration fragments, as mentioned here https://github.com/docker-library/rabbitmq/pull/501#issuecomment-882506021.

Also, do we mean **deprecated** or **removed**? I think it's the latter.

Reply:
I think we're splitting hairs a little bit on the distinction between those two -- we usually use the word "deprecated" for things like that because that's pretty clearly intended to mean "don't use this anymore and it might break in the future" (in this case, the "deprecation period" was just really short and even happened retroactively within the 3.8 series).